subworkflows/local/functional_annotation.nf

```diff
@@ -0,0 +1,233 @@
+    */
+    EGGNOGMAPPER_ORTHOLOGS(
+        ch_protein_chunked,
+        params.eggnog_data_dir,
+        params.eggnog_database,
+        params.eggnog_diamond_database,
+    )
+    ch_versions = ch_versions.mix(EGGNOGMAPPER_ORTHOLOGS.out.versions.first())
+
+    EGGNOGMAPPER_ANNOTATIONS(
+        EGGNOGMAPPER_ORTHOLOGS.out.orthologs,
+        params.eggnog_data_dir,
+    )
+    ch_versions = ch_versions.mix(EGGNOGMAPPER_ANNOTATIONS.out.versions.first())
+
+    CONCATENATE_EGGNOGMAPPER_ORTHOLOGS(
+        EGGNOGMAPPER_ORTHOLOGS.out.orthologs.groupTuple()
+    )
+    ch_versions = ch_versions.mix(CONCATENATE_EGGNOGMAPPER_ORTHOLOGS.out.versions.first())
+
+    CONCATENATE_EGGNOGMAPPER_ANNOTATIONS(
+        EGGNOGMAPPER_ANNOTATIONS.out.annotations.groupTuple()
+    )
+    ch_versions = ch_versions.mix(CONCATENATE_EGGNOGMAPPER_ANNOTATIONS.out.versions.first())
+
+    CONCATENATE_INTERPROSCAN_GFFS(
+        INTERPROSCAN.out.gff3.groupTuple()
+    )
+    ch_versions = ch_versions.mix(CONCATENATE_INTERPROSCAN_GFFS.out.versions)
+
+    /*
+     * Get GO term and GO-slim term counts out of an InterProScan .tsv output file
+     * We ran GOSLIM once per assembly, hence the groupTuple (the IPS results are one per chunk )
+    */
+    GOSLIM_SWF(
+        CONCATENATE_INTERPROSCAN_TSV.out.csv,
+        file(params.go_obo, checkIfExists: true),
+        file(params.goslim_ids, checkIfExists: true),
+        file(params.go_banding, checkIfExists: true),
+    )
+    ch_versions = ch_versions.mix(GOSLIM_SWF.out.versions)
+
+    // TODO: remove this once GOSLIM_SWF produces bgzip/index tsvs
+    TABIX_BGZIP_GO(
+        GOSLIM_SWF.out.go_summary
+    )
+    ch_versions = ch_versions.mix(TABIX_BGZIP_GO.out.versions)
+
+    TABIX_BGZIP_GOSLIM(
+        GOSLIM_SWF.out.goslim_summary
+    )
+    ch_versions = ch_versions.mix(TABIX_BGZIP_GOSLIM.out.versions)
+
+    /*
+     * Assign Rhea and CHEBI tags to the proteins.
+     * This is a 2 step process
+     * 1 - run diamond against a post-processed UniProt90 + Rhea DB
+     * 2 - extract the hits using the mgnif toolkit add rhea annotations script
+    */
+    DIAMOND_RHEACHEBI(
+        ch_proteins_faa,
+        file(params.uniref90rhea_diamond_database, checkIfExists: true),
+        file(params.rheachebi_mapping_tsv, checkIfExists: true)
+    )
+    ch_versions = ch_versions.mix(DIAMOND_RHEACHEBI.out.versions)
+
+    TABIX_BGZIP_RHEAANDCHEBI(
+        DIAMOND_RHEACHEBI.out.rhea2proteins_tsv
+    )
+    ch_versions = ch_versions.mix(TABIX_BGZIP_RHEAANDCHEBI.out.versions)
+
+    ch_proteins_gff.combine(ch_protein_chunked, by: 0).multiMap { meta, gff, faa ->
+        faa: [meta, faa]
+        gff: [meta, gff]
+    }.set {
+        ch_dbcan
+    }
+
+    DBCAN(
+        ch_dbcan.faa,
+        ch_dbcan.gff,
+        [file(params.dbcan_database, checkIfExists: true), params.dbcan_database_version],
+        "protein" // mode
+    )
+    ch_versions = ch_versions.mix(DBCAN.out.versions)
+
+    CONCATENATE_DBCAN_GFFS(
```

Comment:
Maybe this one should be removed and instead the dbCAN GFF produced by GFF Toolkit should be used? I think multiple dbCAN GFFs will make things confusing unless we want to output this one alongside the processed ones.

Reply:
Yeah, I'll review the GFFs when adding the toolkit stuff.